### ISSUE TITLE
Directional Blocks is not placeable

### ISSUE THREAD

**Issue by bkgktp** [oraxen/oraxen#1015]
### I have checked...

- [X] I am using the latest version of Oraxen
- [X] I am using the latest version of any dependencies
- [X] I am using ProtocolLib 5.0
- [X] I have checked if any similar bug reports exist

### Description

When we try to place directional blocks, the console throws an error.

It works fine on old oraxen versions, but after version 1.160.0 we cannot insert faceted blocks.

> 
> palm_log:
>   displayname: <white>Palm Log
>   material: PAPER
>   Pack:
>     generate_model: false
>     model: custom/blocks/logs/palm/palm_log
>     custom_model_data: 1000
>   Mechanics:
>     noteblock:
>       model: custom/blocks/logs/palm/palm_log
>       custom_variation: 101
>       block_sounds:
>         place_sound: block.wood.place
>         break_sound: block.wood.break
>         hit_sound: block.wood.hit
>         step_sound: block.wood.step
>         fall_sound: block.wood.fall
>       logStrip:
>         stripped_log: stripped_palm_log
>         drop: stripped_palm_log
>       directional:
>         directional_type: LOG
>         y_block: palm_log_y
>         x_block: palm_log_x
>         z_block: palm_log_z
>       hardness: 2.0
>       drop:
>         minimal_type: WOOD
>         best_tools:
>         - AXE
>         silktouch: false
>         loots:
>         - oraxen_item: palm_log
>           probability: 1.0
> 
> palm_log_y:
>   excludeFromInventory: true
>   material: PAPER
>   Mechanics:
>     noteblock:
>       custom_variation: 102
>       logStrip:
>         stripped_log: stripped_palm_log_y
>       directional:
>         parent_block: palm_log
> 
> palm_log_x:
>   excludeFromInventory: true
>   material: PAPER
>   Mechanics:
>     noteblock:
>       custom_variation: 103
>       logStrip:
>         stripped_log: stripped_palm_log_x
>       directional:
>         parent_block: palm_log
> 
> palm_log_z:
>   excludeFromInventory: true
>   material: PAPER
>   Mechanics:
>     noteblock:
>       logStrip:
>         stripped_log: stripped_palm_log_z
>       custom_variation: 104
>       directional:
>         parent_block: palm_log
> 
> stripped_palm_log:
>   displayname: <white>Stripped Palm Log
>   material: PAPER
>   Pack:
>     generate_model: false
>     model: custom/blocks/logs/palm/stripped_palm_log
>     custom_model_data: 1001
>   Mechanics:
>     noteblock:
>       model: custom/blocks/logs/palm/stripped_palm_log
>       custom_variation: 105
>       block_sounds:
>         place_sound: block.wood.place
>         break_sound: block.wood.break
>         hit_sound: block.wood.hit
>         step_sound: block.wood.step
>         fall_sound: block.wood.fall
>       directional:
>         directional_type: LOG
>         y_block: stripped_palm_log_y
>         x_block: stripped_palm_log_x
>         z_block: stripped_palm_log_z
>       hardness: 1
>       drop:
>         # minimal_type: WOOD
>         # best_tools:
>         # - AXE
>         silktouch: false
>         loots:
>         - oraxen_item: stripped_palm_log
>           probability: 1.0
> 
> 
> stripped_palm_log_x:
>   excludeFromInventory: true
>   material: PAPER
>   Mechanics:
>     noteblock:
>       custom_variation: 106
>       directional:
>         parent_block: stripped_palm_log
> 
> stripped_palm_log_y:
>   excludeFromInventory: true
>   material: PAPER
>   Mechanics:
>     noteblock:
>       custom_variation: 107
>       directional:
>         parent_block: stripped_palm_log
> 
> stripped_palm_log_z:
>   excludeFromInventory: true
>   material: PAPER
>   Mechanics:
>     noteblock:
>       custom_variation: 108
>       directional:
>         parent_block: stripped_palm_log
> 
> palm_planks:
>   displayname: <white>Palm Planks
>   material: PAPER
>   Pack:
>     generate_model: true
>     parent_model: block/cube_all
>     textures:
>     - custom/blocks/logs/palm/palm_planks.png
>     custom_model_data: 1075
>   Mechanics:
>     noteblock:
>       model: palm_planks
>       custom_variation: 109
>       block_sounds:
>         place_sound: block.wood.place
>         break_sound: block.wood.break
>         hit_sound: block.wood.hit
>         step_sound: block.wood.step
>         fall_sound: block.wood.fall
>       hardness: 2.0
>       drop:
>         # minimal_type: WOOD
>         # best_tools:
>         # - AXE
>         silktouch: false
>         loots:
>         - oraxen_item: palm_planks
>           probability: 1.0
>  
> 
> 
> palm_leaves:
>   displayname: <white>Palm Leaves
>   material: PAPER
>   Pack:
>     generate_model: true
>     parent_model: block/cube_all
>     textures:
>     - custom/blocks/leaves/palm_leaves.png
>     custom_model_data: 1003
> 
> palm_sapling:
>   displayname: <white>Hindistan Cevizi Fidesi
>   material: PAPER
>   Pack:
>     generate_model: true
>     # model: custom/blocks/saplings/palm_sapling
>     parent_model: item/handheld
>     textures:
>     - custom/items/coconut_seedling.png
>     custom_model_data: 1004
> 

> 
> ### 
> [00:22:31] [Server thread/ERROR]: Could not pass event PlayerInteractEvent to Oraxen v1.160.0
> java.lang.NullPointerException: Cannot invoke "io.th0rgal.oraxen.mechanics.provided.gameplay.noteblock.NoteBlockMechanic.getCustomVariation()" because the return value of "io.th0rgal.oraxen.mechanics.provided.gameplay.noteblock.NoteBlockMechanicFactory.getMechanic(String)" is null
> 	at io.th0rgal.oraxen.mechanics.provided.gameplay.noteblock.directional.DirectionalBlock.getDirectionVariation(DirectionalBlock.java:108) ~[oraxen-1.160.0.jar:?]
> 	at io.th0rgal.oraxen.mechanics.provided.gameplay.noteblock.directional.DirectionalBlock.getDirectionVariation(DirectionalBlock.java:72) ~[oraxen-1.160.0.jar:?]
> 	at io.th0rgal.oraxen.mechanics.provided.gameplay.noteblock.NoteBlockMechanicListener.onPrePlacingCustomBlock(NoteBlockMechanicListener.java:331) ~[oraxen-1.160.0.jar:?]
> 	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor294.execute(Unknown Source) ~[?:?]
> 	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77) ~[paper-api-1.20.1-R0.1-SNAPSHOT.jar:?]
> 	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:81) ~[paper-api-1.20.1-R0.1-SNAPSHOT.jar:git-Paper-100]
> 	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-api-1.20.1-R0.1-SNAPSHOT.jar:?]
> 	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[paper-1.20.1.jar:git-Paper-100]
> 	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[paper-1.20.1.jar:git-Paper-100]
> 	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615) ~[paper-api-1.20.1-R0.1-SNAPSHOT.jar:?]
> 	at org.bukkit.craftbukkit.v1_20_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:566) ~[paper-1.20.1.jar:git-Paper-100]
> 	at net.minecraft.server.level.ServerPlayerGameMode.useItemOn(ServerPlayerGameMode.java:537) ~[?:?]
> 	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleUseItemOn(ServerGamePacketListenerImpl.java:1960) ~[?:?]
> 	at net.minecraft.network.protocol.game.ServerboundUseItemOnPacket.handle(ServerboundUseItemOnPacket.java:37) ~[?:?]
> 	at net.minecraft.network.protocol.game.ServerboundUseItemOnPacket.a(ServerboundUseItemOnPacket.java:9) ~[?:?]
> 	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:51) ~[?:?]
> 	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.20.1.jar:git-Paper-100]
> 	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
> 	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
> 	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1339) ~[paper-1.20.1.jar:git-Paper-100]
> 	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:197) ~[paper-1.20.1.jar:git-Paper-100]
> 	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
> 	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1316) ~[paper-1.20.1.jar:git-Paper-100]
> 	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1309) ~[paper-1.20.1.jar:git-Paper-100]
> 	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
> 	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1386) ~[paper-1.20.1.jar:git-Paper-100]
> 	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1171) ~[paper-1.20.1.jar:git-Paper-100]
> 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:318) ~[paper-1.20.1.jar:git-Paper-100]
> 	at java.lang.Thread.run(Thread.java:833) ~[?:?]

### Steps to reproduce

Take a directional block and try to place it horizontally.

### Plugin Version

<img width="385" alt="resim" src="https://github.com/oraxen/oraxen/assets/101702786/6c979d6c-eb12-4cc2-a76b-3a150db83d70">

### ProtocolLib Version

<img width="670" alt="resim" src="https://github.com/oraxen/oraxen/assets/101702786/83832b14-15e1-4c96-abe8-43b0bcfd113b">

### Server Software & Version

<img width="1077" alt="resim" src="https://github.com/oraxen/oraxen/assets/101702786/28154b12-aa3f-469d-a01a-da54c81844d0">

### How breaking is the bug?

Minor Bug - Plugin completely functional, but features have non-working aspects

**Comment by Boy0000** [comment]
1. config issue not a bug
2. Learn to use codeblocks to make ur config examples readable
3. logs go in a pastebin or https://mclo.gs

**Comment by bkgktp** [comment]
https://mclo.gs/HUBNhsi
my server full logs

https://mclo.gs/VICUEjm
item.yml

@Boy0000

**Comment by bkgktp** [comment]
It worked fine in the previous version, why not now?

and also is it related to items or plugin settings?

**Comment by bkgktp** [comment]
After updating the plugin, such errors should not occur.

**Comment by Boy0000** [comment]
as said above, i am entirely unable to replicate this on latest release when following the documentation for Directional Blocks

**Comment by bkgktp** [comment]
https://docs.oraxen.com/mechanics/noteblock-mechanic/directional-mechanic

you're wrong, I'm sure there is a bug in the plugin. It's ridiculous that you override the subject. Even if I do the examples on the wiki, it breaks down. Installing old version oraxen fixes it.

**Comment by Boy0000** [comment]
I am not overriding it...?
I tested it locally, and could not teplicate any issue with it.

You can see if anything changed between the latest version that works for you and the current release

Otherwise ill look whenever i can and can replicate any issues with it

**Comment by bkgktp** [comment]
I told the developer why this is happening but he won't listen to me. Bukkit API is a very sensitive library and can cause problems for small reasons, so the controller should not skip statements. @MarkinhosMLP

**Comment by Boy0000** [comment]
yet again you can say im not listening to you, yet i sent you literal video clip of having fully working directional blocks on latest release... I cannot fix something which im not able to reproduce and noone else seems to have an issue with...

**Comment by bkgktp** [comment]
https://mclo.gs/bKslrZQ

or you can update the DirectionalBlocks section in the plugin wiki like this (logStrip doesn't work or I couldn't get it to work)

**Comment by Boy0000** [comment]
And as I showed you, that is simply not the case. I have fully working Directional Blocks with the current that does not use that.
As I also showed you, isConfiguration section is literally just a fancy getConfiguration != null check

Feel free to open a PR if this actually fixes it for you but thats just weird as me and basically everyone else here dont have this issue. Until then Im gonna assume this is either a bug with your server software or some other random plugin you have.

As always replicate the bug on a **Paper** server with only Oraxen. Some weird fork + tons of other plugins is most often the issue
